lighthouse-core/gather/driver.js

```diff
@@ -418,17 +418,86 @@ class Driver {
     });
   }
 
+  /**
+   * Deprecated: renamed to `evaluate`.
+   * @param {string} expression
+   * @param {{useIsolation?: boolean}=} options
+   * @return {Promise<*>}
+   */
+  async evaluateAsync(expression, options = {}) {
+    return this.evaluate(expression, options);
+  }
+
+  /**
+   * Evaluate an expression in the context of the current page. If useIsolation is true, the expression
+   * will be evaluated in a content script that has access to the page's DOM but whose JavaScript state
+   * is completely separate.
+   * Returns a promise that resolves on the expression's value.
+   * See the documentation for `createEvalCode` in `page-functions.js`.
+   * @template T, R
+   * @param {string | ((...args: T[]) => R)} expressionOrMainFn
+   * @param {{useIsolation?: boolean, args?: T[], deps?: (Function|string)[]}=} options
```

Comment:
FYI I would like to drop the `string` part eventually.

Reply:
> FYI I would like to drop the `string` part eventually.

could we drop it now and say "if you still want to use a string use the legacy `evaluateAsync()`"?

Reply:
done